compose/src/main/java/com/bottlerocketstudios/compose/auth/Login.kt

```diff
@@ -1,16 +1,316 @@
 package com.bottlerocketstudios.compose.auth
 
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.text.KeyboardActions
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.Icon
+import androidx.compose.material.LocalContentColor
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.OutlinedTextField
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.material.TextFieldDefaults
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.State
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.focus.FocusDirection
+import androidx.compose.ui.focus.onFocusChanged
+import androidx.compose.ui.platform.LocalFocusManager
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.input.PasswordVisualTransformation
+import androidx.compose.ui.text.input.VisualTransformation
+import androidx.compose.ui.tooling.preview.Preview
+import com.bottlerocketstudios.compose.R
+import com.bottlerocketstudios.compose.resources.ArchitectureDemoTheme
+import com.bottlerocketstudios.compose.resources.Dimens
+import com.bottlerocketstudios.compose.widgets.OutlinedSurfaceButton
+import com.bottlerocketstudios.compose.widgets.PrimaryButton
+import com.bottlerocketstudios.compose.widgets.SurfaceButton
 
 data class LoginScreenState(
     val email: State<String>,
     val password: State<String>,
     val loginEnabled: State<Boolean>,
     val devOptionsEnabled: Boolean,
     val onLoginClicked: () -> Unit,
+    val onSignupClicked: () -> Unit,
+    val onForgotClicked: () -> Unit,
+    val onDevOptionsClicked: () -> Unit,
+    val onEmailChanged: (String) -> Unit,
+    val onPasswordChanged: (String) -> Unit
 )
 
 @Composable
 fun LoginScreen(state: LoginScreenState) {
+    Surface(
```

Comment:
@ianslinker-br Should the root screen view be some scrolling container to allow for the screen to be scrollable when the soft keyboard is up?

Reply:
In my testing, this still allows the screen to be scrollable when the keyboard is up. The wrapped column content becomes larger than the surface size and the vertical scroll applied to it allows the user to scroll the content as necessary.